code/game/machinery/merch_vendor.dm

```diff
@@ -0,0 +1,126 @@
+/obj/machinery/economy/merch
+	name = "Nanotrasen Merchandise Vendor"
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "nt_merch"
+	light_color = LIGHT_COLOR_GREEN
+
+	var/static/list/merchandise = list()
+	var/static/list/imagelist = list()
+	var/list/merch_categories = list(
+		MERCH_CAT_APPAREL,
+		MERCH_CAT_TOY,
+		MERCH_CAT_DECORATION,
+	)
+
+/obj/machinery/economy/merch/attack_ai(mob/user as mob)
+	return attack_hand(user)
+
+/obj/machinery/economy/merch/attack_hand(mob/user)
+	ui_interact(user)
+
+/obj/machinery/economy/merch/Initialize(mapload)
+	. = ..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/merch(null)
+	if(!length(merchandise))
+		for(var/merch_path in subtypesof(/datum/merch_item))
+			var/datum/merch_item/merch = new merch_path()
+			merchandise[merch.category] += list(merch)
+			var/obj/item/I = merch.typepath
+			var/pp = replacetext(replacetext("[merch.typepath]", "/obj/item/", ""), "/", "-")
+			imagelist[pp] = "[icon2base64(icon(initial(I.icon), initial(I.icon_state), SOUTH, 1))]"
+
+/obj/machinery/economy/merch/attackby(obj/item/I, mob/user)
+	if(isspacecash(I))
+		insert_cash(I, user)
+		return TRUE
+
+	return ..()
+
+/obj/machinery/economy/merch/proc/do_purchase(datum/money_account/customer_account, datum/merch_item/merch, mob/user)
+	if(!merch)
+		return FALSE
+
+	if(cash_stored >= merch.cost)
+		pay_with_cash(merch.cost, "Purchase of [merch.name]", name, user, account_database.vendor_account)
+		give_change(user)
+	else
+		var/mob/living/carbon/human/H = user
+		if(istype(H))
+			var/obj/item/card/id/C = H.get_idcard(TRUE)
+			if(!C || !pay_with_card(C, merch.cost, "Purchase of [merch.name]", "NAS Trurl Merchandising", user, account_database.vendor_account))
+				return FALSE
+		else
+			to_chat(user, "<span class='warning'>Payment failure: you have no ID or other method of payment.</span>")
+			return FALSE
```

Comment:
Optionally, this can be put in its own proc `attempt_transaction(...)` or something

Reply:
done in next commit